### PR TITLE
Add processor that sanitizes configurable keys

### PR DESCRIPTION
**Pull request by rz**
The logic of the new processor is almost identical to the logic of the
existing SanitizePasswordsProcessor, so this commit also changes the
implementation of SanitizePasswordsProcessor to avoid code-duplication.
It now inherits from the new SanitizeKeysProcessor.

### CODE REVIEW

**Comment by rz** [comment]
@mattrobenolt this is my shot at implementing #984.

i think a test to ensure the config is passed-in correctly in all the scenarios may be in order, but i didn't see an obvious place for that. otoh, that change is literally 1 line of code, adding 1 attribute to the Client on init, so maybe it is ok to do without the test.

**Comment by rz** [comment]
@mattrobenolt it would be very awesome if this could make it into an official release soon so that i can install the official version in production for work.

**Comment by ehfeng** [comment]
If we merge this, ping me and I'll add this to our docs.

**Comment by mattrobenolt** [comment]
> If we merge this

and tag a release.

File: raven/processors.py [hunk 1]
```diff
@@ -148,3 +140,31 @@ def _sanitize_keyvals(self, keyvals, delimiter):
...
+        'api_key',
+        'apikey',
+        'sentry_dsn',
+        'access_token',
+    ])
+    VALUES_RE = re.compile(r'^(?:\d[ -]*?){13,16}$')
+
+    def __init__(self, client):
+        super(SanitizeKeysProcessor, self).__init__(client)  # run the __init__ method of Processor, not SanitizeKeysProcessor
+        self.fields = self.FIELDS
```
Comment (cancan101) [inline raven/processors.py]: Seems odd to rebind field.

File: raven/processors.py [hunk 2]
```diff
@@ -66,31 +66,23 @@ def filter_stacktrace(self, data, **kwargs):
...
-        'api_key',
-        'apikey',
-        'sentry_dsn',
-        'access_token',
-    ])
-    VALUES_RE = re.compile(r'^(?:\d[ -]*?){13,16}$')
+
+    def __init__(self, client):
+        super(SanitizeKeysProcessor, self).__init__(client)
+        fields = getattr(client, 'sanitize_keys')
```
Comment (ashwoods) [inline raven/processors.py]: Could you move this into a `get_fields` method, or rather `get_keys` to be more consistent with naming.

File: raven/processors.py [hunk 3]
```diff
@@ -102,7 +94,7 @@ def sanitize(self, key, value):
             key = text_type(key)
 
         key = key.lower()
-        for field in self.FIELDS:
+        for field in self.fields:
```
Comment (ashwoods) [inline raven/processors.py]: this would change to `for field in self.get_keys()`

File: raven/processors.py (hunk 1 above)
Comment (ashwoods) [inline raven/processors.py]: and then it wouldn't be necessary to rebind. just override `get_keys`.

**Pull request closed without merging**